Sources/MapboxCoreNavigation/PredictiveCache/PredictiveCacheOptions.swift

```diff
@@ -0,0 +1,91 @@
+import MapboxNavigationNative
+import MapboxDirections
+/**
+ Specifies the content that a predictive cache fetches and how it fetches the content.
+ 
+ Pass an instance of this class into the `NavigationOptions(styles:navigationService:voiceController:topBanner:bottomBanner:predictiveCacheOptions:)` initializer or `NavigationMapView.enablePredictiveCaching(options:)` method.
+ */
+public struct PredictiveCacheOptions {
+
+    /**
+     Predictive cache Navigation related options
+    */
+    public var predictiveCacheNavigationOptions: PredictiveCacheNavigationOptions = .init()
+
+    /**
+     Predictive cache Map related options
+     */
+    public var predictiveCacheMapsOptions: PredictiveCacheMapsOptions = .init()
+    
+    /**
+     How far around the user's location caching is going to be performed.
+     
+     Defaults to 2000 meters.
+     */
+    @available(*, deprecated, message: "Use `predictiveCacheNavigationOptions` and `predictiveCacheMapsOptions` instead.")
```

Comment:
Not sure, maybe it should not be marked as deprecated in order to simplify the case when users need the same options for maps and navigation

Reply:
I think that deprecation makes sense here because old implementation assumes that both values are the same, but if it's not, `getter` will become incorrect. And it is also a good practice to point developers to use recommended setup.